### DISCUSSION TITLE
Long pauses during inbound SMTP transactions

### DISCUSSION THREAD

**Discussion by Ramon-LibertyPumps** [postalserver/postal#1956] [Help with using Postal] [answered]
I'm seeing long delays when servers from the Internet connect to my postal server smtp-server to deliver mail. Below is a sample. I've inserted blank lines where I see two long delays: one after receiving a EHLO and another after sending capabilities.

[SMTP-1.txt](https://github.com/postalserver/postal/files/8489960/SMTP-1.txt)

Here is another example with 3 pauses. This server gives up and disconnects at the end.

[SMTP-2.txt](https://github.com/postalserver/postal/files/8489959/SMTP-2.txt)

CPU usage is very low during the whole thing. My docker-compose for the smtp service looks like this:

  smtp:
    container_name: postal_smtp
    image: ghcr.io/postalserver/postal:2.1.1
    restart: unless-stopped
    depends_on:
      - mariadb
      - rabbitmq
    command: postal smtp-server
    network_mode: host
    cap_add:
      - NET_BIND_SERVICE
      - NET_ADMIN
    volumes:
      - /opt/postal/config:/config
      - /opt/postal/log:/opt/postal/app/log
      - /opt/postal/certificates:/certificates

postal.yml -
smtp_server:
  \# Specify configuration to the Postal SMTP server
  port: 25
  tls_enabled: false
\#  tls_certificate_path: /certificates/wildcard_.mydomain.com/wildcard_.mydomain.com.crt
\#  tls_private_key_path: /certificates/wildcard_.mydomain.com/wildcard_.mydomain.com.key
  log_connect: true
  strip_received_headers: false

Any tips that can lead me in the right direction would be greatly appreciated.

Thanks.

**Comment by Ramon-LibertyPumps** [comment, marked as answer]
So, this turned out to have a simple resolution.

In the smtp_server section, I set the bind_address to 0.0.0.0. This got rid of the IPv6 prefixes in front of all the IP addresses, and made everything snappy. I'm thinking the slowdowns maybe happen during DNS lookups or something. I don't know anything about Ruby, though, so I can't tell if the fix would be in postal or some other component that it uses.

**Reply by willpower232** [reply]
glad you figured it out, its possible that something either on your server or with your server provider isn't quite set up for resolving it
